### DISCUSSION TITLE
Why is the button in the modal component wrapped in a `<span>`?

### DISCUSSION THREAD

**Discussion by hirasso** [alpinejs/alpine#4048] [1. Help] [answered]
Hi there! 

I just noticed this bit in the [modal component's example code](https://alpinejs.dev/component/modal):

```html
<span x-on:click="open = true">
    <button type="button" class="rounded-md bg-white px-5 py-2.5 shadow">
        Open dialog
    </button>
</span>
```

I was wondering what's the reason the button is wrapped in a `<span>` here, instead of putting `x-on:click` directly on the button itself? That's also how it used to be, as you can see in the video:

![CleanShot 2024-02-23 at 10 28 08@2x](https://github.com/alpinejs/alpine/assets/869813/f427ff2c-3fc2-4308-b889-b8b94312f060)

So there must be some reason (a11y??) that the wrapping span was added sometime after the video was recorded. 

Any insight would be much appreciated.

**Comment by ekwoka** [comment, marked as answer]
There is none.

It really shouldn't be there.

I feel it's most likely a case of layout being adjusted while doing some styling of the component for the documentation and pieces were moved around and by the end it was not then refactored/reduced.

**Reply by hirasso** [reply]
Thanks for your answer! Is the components website open source somewhere to create a PR with a fix? I guess not as it's not accessible for non-sponsors...

If not, is @calebporzio reachable somehow without needing to use X?
